src/authentication/auth.js

```diff
@@ -0,0 +1,54 @@
+const jwt = require("jsonwebtoken");
+const fs = require("fs");
+const path = require("path");
+const axios = require("axios");
+
+const usersPath = path.join(__dirname, "/users.json");
+
+const secretKey = "secret-key";
+const users = JSON.parse(fs.readFileSync(usersPath));
+
+//Handler for the {domain}/login route. Currently implemented with hardcoded user credentials (users.json) and observed via Postman
+const login = async (req, res) => {
+  const { username, password } = req.body;
+
+  const user = users.find((user) => user.username === username);
+
+  if (!user) {
+    return res.status(401).json({ message: "Invalid username or password" });
+  }
+
+  //Create token
+  if (password === user.password) {
+    const token = jwt.sign(
+      { sub: user.id, username: user.username },
+      secretKey,
+      { expiresIn: "1h" }
+    );
+
+    res.json({ token: token, message: "Token created successfully" });
+
+    //Include token in HTTP headers
+
+    req.header["Authorization"] = token;
```

Comment:
You don't have to include it in the header when you respond back to client.
Also, if you want to include it, you should include it in the response object `res`, not `req`. This method should be an endpoint, not a middleware function.

Reply:
I don't understand this comment, specifically the last sentence

Reply:
Is the `login` function a middleware function or an endpoint?

The answer is obviously that the function is an **endpoint**, hence the response of this function will be sent to the http client (user). In the case of a middleware, the response is sent to the _next_ function, that is why we use the `next()` call.

So, when you are doing an HTTP response,  the http protocol sends RESPONSE headers, not REQUEST headers.
So to say, the req.header["Authorization"] = token; is meaningless in the context of this function.

Reply:
To better understand this, is to keep in mind that HTTP is a stateless protocol, and what you do in this function, has no impact on the client side. The only thing that COULD have an impact is the response you are sending.